pkg/client/didexchange/client.go

```diff
@@ -259,3 +259,19 @@ func (c *Client) handleMessageEvent(msg *service.StateMsg) {
 		handler <- *msg
 	}
 }
+
+// EventProperties returns the DIDExchange protocol related data along with any processing errors. Processing error
+// will have value only in case of abandoned state.
+func EventProperties(e service.EventProperties) (didexchange.Event, error, error) {
```

Comment:
Do not do this `(didexchange.Event, error, error)` only one error should be returned

Reply:
Changed the type of second return type to [service.ErrEventProcess]( https://github.com/hyperledger/aries-framework-go/compare/3c7cd17e91d97a198ad9fbe98046c1373efc7d6c..d215a24d2cb11e4f38c59581ccdcb6a73fc7d364#diff-427522b353a28b1ec350c29c79d387b0R271). 

I didn't add to combine both errors into one (ultimately consumer can switch based on error type) as consumer will ignore the first type (didexchange event props - connectionID and invitationID). Here user can check for error and then process the logic based on correlation in didexchange.Event and service.ErrEventProcess(if any - In case of state type is abandoned).

Reply:
I still do not like it, I would rather split it into two different functions. The logic here has two different behavior what is a bit confused.

Reply:
Removed this util function as consumer need to do check.